ha/util/conf_store.py

```diff
@@ -221,3 +227,25 @@ def get_disk_list_for_cvg(index, node_id, cvg_id):
         except Exception as e:
             Log.error(f"Unable to fetch Disk list from GConf. Error {e}")
             raise Exception(f"Unable to fetch Disk list. Error {e}")
+
+
+    @staticmethod
+    def get_cvg_for_disk(node_id, disk_id):
+        """
+        Returns cvg id for given node_id and disk_id.
+        Args:
+            index(str): index of conf file
+            node_id (str): node id
+            disk_id (str): disk id
+        Returns:
+            cvg id in string format
+
+        >>> ConftStoreSearch.get_cvg_for_disk('21d6291109304485b3daff43a06cff77', '/dev/sdd')
+        'cvg-01'
+        """
+        cvg_id = ""
+        cvg_list = ConftStoreSearch.get_cvg_list(CLUSTER_GLOBAL_INDEX, node_id)
+        for cvg_id in cvg_list:
+            if disk_id in ConftStoreSearch.get_disk_list_for_cvg(CLUSTER_GLOBAL_INDEX, node_id, cvg_id):
+                return cvg_id
+        return cvg_id
```

Comment:
cvg_id should not be returned here right  ? it should be returned only on line 250 I guess

Reply:
It would return empty string in case not entered into loop.
But fine to just mention only inside the loop.

Reply:
Thanks. It would not return empty string if cvg_id is valid but disk id is not present inside this cvg_id and disk-id is not present in any of the cvg_ids. In this case incorrect cvg)id would be returned